=== modified file 'configure.ac'
--- configure.ac	2009-12-10 19:18:44 +0000
+++ configure.ac	2009-12-11 14:17:10 +0000
@@ -16,6 +16,7 @@
 AC_PROG_LIBTOOL
 
 AC_SUBST(VERSION)
+AC_SUBST(DOC_SERVER)
 AC_CONFIG_MACRO_DIR([m4])
 
 m4_ifdef([AM_SILENT_RULES],[AM_SILENT_RULES([yes])])
@@ -93,6 +94,7 @@
 docs/Makefile
 docs/reference/Makefile
 docs/reference/version.xml
+docs/reference/libappindicator-docs.sgml
 ])
 
 ###########################

=== modified file 'docs/reference/Makefile.am'
--- docs/reference/Makefile.am	2009-12-08 22:02:28 +0000
+++ docs/reference/Makefile.am	2009-12-11 14:09:14 +0000
@@ -83,7 +83,7 @@
 
 # Other files to distribute
 # e.g. EXTRA_DIST += version.xml.in
-EXTRA_DIST += version.xml.in
+EXTRA_DIST += version.xml.in libappindicator-docs.sgml.in
 
 # Files not to distribute
 # for --rebuild-types in $(SCAN_OPTIONS), e.g. $(DOC_MODULE).types

=== renamed file 'docs/reference/libappindicator-docs.sgml' => 'docs/reference/libappindicator-docs.sgml.in'
--- docs/reference/libappindicator-docs.sgml	2009-12-08 19:22:45 +0000
+++ docs/reference/libappindicator-docs.sgml.in	2009-12-11 14:16:59 +0000
@@ -8,14 +8,14 @@
   <bookinfo>
     <title>libappindicator Reference Manual</title>
     <releaseinfo>
-      for libappindicator [VERSION]
+      for libappindicator @VERSION@
       The latest version of this documentation can be found on-line at
-      <ulink role="online-location" url="http://[SERVER]/libappindicator/index.html">http://[SERVER]/libappindicator/</ulink>.
+      <ulink role="online-location" url="http://@DOC_SERVER@/libappindicator/current/index.html">http://@DOC_SERVER@/libappindicator/current/</ulink>.
     </releaseinfo>
   </bookinfo>
 
   <chapter>
-    <title>[Insert title here]</title>
+    <title>Base Classes for Applications</title>
         <xi:include href="xml/app-indicator.xml"/>
 
   </chapter>

