=== modified file 'src/app-indicator.c'
Index: libappindicator.ubu/src/app-indicator.c
===================================================================
--- libappindicator.ubu.orig/src/app-indicator.c	2011-02-28 08:35:28.963390000 -0500
+++ libappindicator.ubu/src/app-indicator.c	2011-02-28 08:46:54.147065973 -0500
@@ -667,6 +667,8 @@
 		priv->connection = NULL;
 	}
 
+	g_signal_handlers_disconnect_by_func(gtk_icon_theme_get_default(), G_CALLBACK(theme_changed_cb), self);
+
 	G_OBJECT_CLASS (app_indicator_parent_class)->dispose (object);
 	return;
 }
